source/s3_client.c

```diff
@@ -343,31 +343,10 @@ struct aws_s3_client *aws_s3_client_new(
         }
     }
 
-    client->buffer_pool = aws_s3_buffer_pool_new(allocator, part_size, mem_limit);
-
-    if (client->buffer_pool == NULL) {
-        goto on_early_fail;
-    }
```

Comment:
do we still need to move any of this code around? Can we keep it where it was, and change ~goto on_early_fail~ -> `goto on_error` and call it a day?

Reply:
Yeah, we need to make sure we acquire the bootstrap reference before `on_error`, since `on_error` releases the reference.

Reply:
Thanks, updated with a test.